frontends/web/src/connected.tsx

```diff
@@ -14,46 +15,28 @@
  * limitations under the License.
  */
 
-import { Component, ReactNode } from 'react';
+import { ReactNode, useEffect, useState } from 'react';
 import { backendConnected } from './api/subscribe';
 
-interface State {
-  connected: boolean;
+type TProps = {
+    children: ReactNode;
 }
 
-interface Props {
-  children: ReactNode;
-}
-
-class ConnectedApp extends Component<Props, State> {
-  public readonly state: State = {
-    connected: true,
-  };
-
-  private unsubscribe!: () => void;
-
-  public componentDidMount() {
-    this.unsubscribe = backendConnected(connected => this.setState({ connected }));
-  }
+export const ConnectedApp = ({ children }: TProps) => {
+  const [connected, setConnected] = useState(true);
 
-  public componentWillUnmount() {
-    this.unsubscribe();
-  }
+  useEffect(() => {
+    return backendConnected(connected => {
+      setConnected(connected);
+    });
+  }, []);
```

Comment:
this is shorthard and returns the unsibscibe function directly. In most other places we store it in a variable first `const unsubscribe = ` and the `return unsubscribe;`.

Note this could even be written shorted as: `useEffect(() => backendConnected(setConnected));` but it is not very obvious that there is an unsubscribe.

I personally think it's ok like this. cc @Beerosagos any preference?

Reply:
I think easy to read code is important but with react effect hooks you kind of need to know how they work even to understand the more verbose version, so I think the even shorter version you suggested is best (no strong opinion).

nit: should have `[]` dependency array, otherwise it will re-run every time the component is re-rendered which is not really necessary (but would also work)

Reply:
Note: I don't want to propose shorter version, but rather get feedback if we should use longer version for readability:

```js
useEffect(() => {
  const unsubscribe = backendConnected(connected => {
    setConnected(connected);
  });
  return unsubscribe;
}, []);
```

Reply:
> I personally think it's ok like this. cc @Beerosagos any preference?

No strong opinion, either. Since there is the `return` statement I think it is readable enough :+1: